gradebook.cpp

```diff
@@ -137,17 +139,104 @@ double Gradebook::get_category_grade(std::string category){
         return this->Get_Exam_Grade();
     }
     else {
-	    return -1.0;
+        return -1.0;
     }
     return 0;
 }
 
 // Return the total cumulative grade for the class as a percentage
 double Gradebook::get_total_grade(){
-    return 0;
+    int tot_points = 1000;
+    double tot_grade = this->Get_Assignment_Grade()+ this->Get_Exam_Grade() + this->Get_Lab_Grade() + this->Get_Projects_Grade();
+    return (tot_grade/tot_points)*100;
 }
 
 // Manually add a new grade to the respective category
-void Gradebook::add_grade(std::string category,std::string name, double grade){
-
-}
+void Gradebook::add_grade(std::string category,std::string name, double grade,std::string fname){

```

Comment:
I'm not sure exactly what's being done here. Why are the names of each assignment being compared? It seems to me like they could just be pushed back into the vector, seeing as the order of the assignments won't affect the grade calculations.

Reply:
You're right it won't affect the calculations, but so that when grades are inputted into the file, they are organized by first, second, third instead of it being 1,3,2 or something